pymove/tests/core/test_data_frame.py

```diff
@@ -0,0 +1,1102 @@
+    )
+    assert_frame_equal(new_move_df, expected)
+    assert isinstance(new_move_df, PandasMoveDataFrame)
+    assert WEEK_END not in move_df
+
+    move_df.generate_weekend_features()
+    assert_frame_equal(move_df, expected)
+
+
+def test_generate_time_of_day_features():
+    move_df = _default_move_df()
+
+    new_move_df = move_df.generate_time_of_day_features(inplace=False)
+    expected = DataFrame(data=[
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1, 'Early morning'],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1, 'Early morning'],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, 'Early morning'],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, 'Early morning'],
+    ], columns=['lat', 'lon', 'datetime', 'id', 'period'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(new_move_df, expected)
+    assert isinstance(new_move_df, PandasMoveDataFrame)
+    assert PERIOD not in move_df
+
+    move_df.generate_time_of_day_features()
+    assert_frame_equal(move_df, expected)
+
+
+def test_generate_datetime_in_format_cyclical():
+    move_df = _default_move_df()
+
+    new_move_df = move_df.generate_datetime_in_format_cyclical(inplace=False)
+    expected = DataFrame(data=[
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1, 0.9790840876823229,
+         0.20345601305263375],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1, 0.9790840876823229,
+         0.20345601305263375],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, 0.9790840876823229,
+         0.20345601305263375],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, 0.9790840876823229,
+         0.20345601305263375],
+    ], columns=['lat', 'lon', 'datetime', 'id', 'hour_sin', 'hour_cos'],
+        index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(new_move_df, expected)
+    assert isinstance(new_move_df, PandasMoveDataFrame)
+    assert HOUR_SIN not in move_df
+
+    move_df.generate_datetime_in_format_cyclical()
+    assert_frame_equal(move_df, expected)
+
+
+def test_generate_dist_time_speed_features():
+    move_df = _default_move_df()
+
+    new_move_df = move_df.generate_dist_time_speed_features(inplace=False)
+    expected = DataFrame(data=[
+        [1, 39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), nan, nan, nan],
+        [1, 39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 13.690153134343689,
+         1.0, 13.690153134343689],
+        [2, 39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), nan, nan, nan],
+        [2, 39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 0.0, 0.0, nan],
+    ], columns=['id', 'lat', 'lon', 'datetime', 'dist_to_prev', 'time_to_prev',
+                'speed_to_prev'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(new_move_df, expected)
+    assert isinstance(new_move_df, PandasMoveDataFrame)
+    assert DIST_TO_PREV not in move_df
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574862)